### PR TITLE
feat/examples: update and revive email.rs examples

### LINKED ISSUE #772: Restore deleted examples and use latest APIs
We previously deleted the files for examples that had been commented out, namely email.rs and email_stress_test.rs. These files should be restored from version control, uncommented and reworked so that they function correctly.

One issue here though is that we need to create an app client through registered_with_keys, i.e. the app needs to be authenticated first. So we either need to use system_uri in the examples, or run them on the mock-routing version only.

Also get rid of docopt and rustc-serialize for command line args parsing and completely move to clap + serde in the examples.

### PR DESCRIPTION
**Pull request by Yoga07**
Rebased on top of #1156 
- Updates and revives email.rs example - real-network only
- Also restores the email_stress_test.rs example which is yet to be updated

### CODE REVIEW

File: safe_app/examples/email.rs [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+// For explanation of lint checks, run `rustc -W help`.
+#![warn(
+    missing_docs,
+    trivial_casts,
+    trivial_numeric_casts,
+    unused_extern_crates,
+    unused_import_braces,
+    unused_qualifications,
+    unused_results
+)]
```
[the remark this answers is not in the document]
Comment (Yoga07) [reply safe_app/examples/email.rs]: `safe_core`'s lint rules are used except for `#![deny(unsafe_code)]`(as we have unsafe ffi blocks in this example). And I think safe_app and safe_auth follow the same as safe_core's.

File: safe_app/Cargo.toml [hunk 2]
```diff
@@ -61,3 +61,8 @@ name = "client_stress_test"
 [[example]]
 bench = false
 name = "self_authentication"
+required-features = ["mock-network"]
```
Comment (lionel1704) [inline safe_app/Cargo.toml]: Why is this a required feature?

Comment (Yoga07) [reply safe_app/Cargo.toml]: Ah, that shouldn't be there. Wrote it to test a few things. Removing it now. Thanks!

**Comment by lionel1704** [comment]
The email example will make more sense once we have the "Sequence" and "Map" types implemented.
Closing this for now.

**Pull request closed without merging**